kubernetes/istio.go

```diff
@@ -0,0 +1,663 @@
+package kubernetes
```

Comment:
I haven't carefully reviewed all the business logic here but I guess this result from moving code without functional changes, right?

Reply:
Right. There are a couple of functional changes: adding Instances/Handlers as plain Istio config. But most of the refactor is just to simplify the kubernetes package using shared methods when possible.

Also a couple of renaming changes that were always there and I think it makes sense to simplify for better separation of concerns between the package.